components/eamxx/src/share/grid/remap/vertical_remapper.cpp

```diff
@@ -0,0 +1,286 @@
+  const auto src_grid = get_src_grid();
+  const auto tgt_grid = get_tgt_grid();
+  if (m_comm.am_i_root()) {
+    printf("-------------------------------------\n");
+    printf("  Setup for vertical remapper\n");
+    printf("  ---------------------------\n");
+    // Print the remap layout
+    printf("  Layout of src->tgt:  (%d, %d) -> (%d, %d)\n",src_grid->get_num_global_dofs(),src_grid->get_num_vertical_levels(),
+                  tgt_grid->get_num_global_dofs(),tgt_grid->get_num_vertical_levels());
+    // Print the set of fields to be mapped
+    printf("  ---------------------------\n");
+    printf("  Map pairs for %d fields:\n",m_num_fields);
+    for (int i=0; i<m_num_fields; ++i) {
+      const auto& f_src  = m_src_fields[i];
+      const auto& f_tgt  = m_tgt_fields[i];
+      printf("      %s -> %s\n",f_src.name().c_str(),f_tgt.name());
+    }
+    printf("-------------------------------------\n");
+  }
+}
+
+FieldLayout VerticalRemapper::
+create_src_layout (const FieldLayout& tgt_layout) const
+{
+  using namespace ShortFieldTagsNames;
+  const auto lt = get_layout_type(tgt_layout.tags());
+  auto src = FieldLayout::invalid();
+  const bool midpoints = tgt_layout.has_tag(LEV);
+  const int vec_dim = tgt_layout.is_vector_layout() ? tgt_layout.dim(CMP) : -1;
+  switch (lt) {
+    case LayoutType::Scalar2D:
+      src = m_src_grid->get_2d_scalar_layout();
+      break;
+    case LayoutType::Vector2D:
+      src = m_src_grid->get_2d_vector_layout(CMP,vec_dim);
+      break;
+    case LayoutType::Scalar3D:
+      src = m_src_grid->get_3d_scalar_layout(midpoints);
+      break;
+    case LayoutType::Vector3D:
+      src = m_src_grid->get_3d_vector_layout(midpoints,CMP,vec_dim);
+      break;
+    default:
+      EKAT_ERROR_MSG ("Layout not supported by VerticalRemapper: " + e2str(lt) + "\n");
+  }
+  return src;
+}
+FieldLayout VerticalRemapper::
+create_tgt_layout (const FieldLayout& src_layout) const
+{
+  using namespace ShortFieldTagsNames;
+  const auto lt = get_layout_type(src_layout.tags());
+  auto tgt = FieldLayout::invalid();
+  const bool midpoints = src_layout.has_tag(LEV);
+  const int vec_dim = src_layout.is_vector_layout() ? src_layout.dim(CMP) : -1;
+  switch (lt) {
+    case LayoutType::Scalar2D:
+      tgt = m_tgt_grid->get_2d_scalar_layout();
+      break;
+    case LayoutType::Vector2D:
+      tgt = m_tgt_grid->get_2d_vector_layout(CMP,vec_dim);
+      break;
+    case LayoutType::Scalar3D:
+      tgt = m_tgt_grid->get_3d_scalar_layout(midpoints);
+      break;
+    case LayoutType::Vector3D:
+      tgt = m_tgt_grid->get_3d_vector_layout(midpoints,CMP,vec_dim);
+      break;
+    default:
+      EKAT_ERROR_MSG ("Layout not supported by VerticalRemapper: " + e2str(lt) + "\n");
+  }
+  return tgt;
+}
+
+void VerticalRemapper::
+set_pressure_levels(const std::string& map_file) {
+  // TODO: I have concerns here with packsize.  I am using SCREAM_PACK_SIZE, which appears to work
```

Comment:
You need to pick a packsize that you're ok with. Then, during the do_remap method, you should check to see if the src/tgt fields are compatible with your pack size. That's what we do in the PD remapper. Something like

```
  if (src and tgt compatible with packsize=N1)
     call_my_remap<N1>(...);
   else if (src and tgt compatible with packsize N2)
     call_my_remap<N2>(...);
   ...
   else
     call_my_remap<1>();
```
You decide how many pack sizes you want to support. The easiest choice is to support N=SCREAM_PACK_SIZE and N=1.

Reply:
It looks like you are only using SCREAM_PACK_SIZE. Are you ok not supporting any other pack size (or even packsize=1, i.e., no packing)?